### ISSUE TITLE
Spotinst: Unable to do rolling-update after upgrading to Kops 1.24.0

### ISSUE THREAD

**Issue by ReillyBrogan** [kubernetes/kops#13964]
/kind bug

**1. What `kops` version are you running? The command `kops version`, will display
 this information.**
Client version: 1.24.0 (git-v1.24.0)

**2. What Kubernetes version are you running? `kubectl version` will print the
 version if a cluster is running or provide the Kubernetes version specified as
 a `kops` flag.**
1.23.8

**3. What cloud provider are you using?**
AWS/Spotinst
`export KOPS_FEATURE_FLAGS="+Spotinst,+SpotinstOcean"`

**4. What commands did you run?  What is the simplest way to reproduce this issue?**
kops rolling-update cluster --instance-group-roles master,nodes --validation-timeout 20m --yes

**5. What happened after the commands executed?**
```
I0711 11:20:46.043441   15757 featureflag.go:162] FeatureFlag "Spotinst"=true
I0711 11:20:46.043482   15757 featureflag.go:162] FeatureFlag "SpotinstOcean"=true
Using cluster from kubectl context: k8s.our.cluster

NAME                    STATUS          NEEDUPDATE      READY   MIN     TARGET  MAX     NODES
master-us-east-2a       NeedsUpdate     1               0       1       1       1       1
master-us-east-2b       NeedsUpdate     1               0       1       1       1       1
master-us-east-2c       NeedsUpdate     1               0       1       1       1       1
nodes                   NeedsUpdate     5               0       0       0       0       5
I0711 11:20:49.717739   15757 instancegroups.go:470] Validating the cluster.
I0711 11:20:51.001961   15757 instancegroups.go:503] Cluster validated.
I0711 11:20:51.002005   15757 instancegroups.go:400] Draining the node: "ip-10-152-1-233.us-east-2.compute.internal".
panic: interface conversion: interface {} is *spotinst.awsElastigroupInstanceGroup, not *autoscaling.Group

goroutine 247 [running]:
k8s.io/kops/upup/pkg/fi/cloudup/awsup.deregisterInstance({0x504cfc0?, 0xc000d94c60?}, 0x3500?)
        k8s.io/kops/upup/pkg/fi/cloudup/awsup/aws_cloud.go:581 +0x4f1
k8s.io/kops/upup/pkg/fi/cloudup/awsup.(*awsCloudImplementation).DeregisterInstance(0xc00171a000?, 0xc000361980?)
        k8s.io/kops/upup/pkg/fi/cloudup/awsup/aws_cloud.go:548 +0x5d
k8s.io/kops/pkg/instancegroups.(*RollingUpdateCluster).drainNode(0xc00171a000, 0xc000e44380)
        k8s.io/kops/pkg/instancegroups/instancegroups.go:648 +0x2c2
k8s.io/kops/pkg/instancegroups.(*RollingUpdateCluster).drainTerminateAndWait(0xc00171a000, 0xc000e44380, 0xc000135d00?)
        k8s.io/kops/pkg/instancegroups/instancegroups.go:402 +0x19a
k8s.io/kops/pkg/instancegroups.(*RollingUpdateCluster).rollingUpdateInstanceGroup.func1(0xc00059001c?)
        k8s.io/kops/pkg/instancegroups/instancegroups.go:193 +0x30
created by k8s.io/kops/pkg/instancegroups.(*RollingUpdateCluster).rollingUpdateInstanceGroup
        k8s.io/kops/pkg/instancegroups/instancegroups.go:192 +0xcca
```
**6. What did you expect to happen?**
A rolling update to happen

**Comment by ReillyBrogan** [comment]
Looking through the source files mentioned in the stacktrace it looks like #12983 (which added `deregisterInstance`) is the likely culprit here. CC @zetaab as author of that MR and @liranp as the maintainer of the Spotinst code.

**Comment by ReillyBrogan** [comment]
Confirmed fixed by #13970, thanks @hakman !
